CRM/Report/Form/Contribute/Detail.php

```diff
@@ -577,6 +563,33 @@ public function postProcess() {
     $this->endPostProcess($rows);
   }
 
+  /**
+   * Shared function for preliminary processing.
+   *
+   * This is called by the api / unit tests and the form layer and is
+   * the right place to do 'initial analysis of input'.
+   */
+  public function beginPostProcessCommon() {
+    // CRM-18312 - display soft_credits and soft_credits_for column
+    // when 'Contribution or Soft Credit?' column is not selected
+    if (empty($this->_params['fields']['contribution_or_soft'])) {
+      $this->_params['fields']['contribution_or_soft'] = 1;
+      $this->noDisplayContributionOrSoftColumn = TRUE;
+    }
+
+    if (CRM_Utils_Array::value('contribution_or_soft_value', $this->_params) ==
+      'contributions_only' &&
+      !empty($this->_params['fields']['soft_credit_type_id'])
+      || !empty($this->_params['soft_credit_type_id_value'])
```

Comment:
Seems the `||` condition need to be inside brackets? Currently, the status message is shown if the soft credit type is selected in the filter with `Contribution OR Soft Credit?` set to any value. Works fine with below if statement -

    if (CRM_Utils_Array::value('contribution_or_soft_value', $this->_params) ==
      'contributions_only' &&
      (!empty($this->_params['fields']['soft_credit_type_id'])
      || !empty($this->_params['soft_credit_type_id_value']))

Reply:
Thanks @jitendrapurohit  - I just pushed a fix